### PR TITLE
enable and fixes errors for `@typescript-eslint/no-floating-promises` rule

### PR DESCRIPTION
**Pull request by B2o5T**
fixes https://github.com/kamilkisiela/graphql-hive/issues/148

### CODE REVIEW

File: packages/libraries/client/src/apollo.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export function hiveApollo(clientOrOptions: HiveClient | HivePluginOptions): Apo
         },
       });
 
-  hive.info();
+  void hive.info();
```
Comment (B2o5T) [inline packages/libraries/client/src/apollo.ts:42]: I think in this line we don't need to await, in cases like this `void` operator can be used [more info on the docs](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-floating-promises.md#ignorevoid)

File: packages/services/api/src/modules/shared/providers/message-bus.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import Emittery from 'emittery';
 export class MessageBus {
   private emitter = new Emittery();
 
-  async on<TPayload>(event: string, listener: (payload: TPayload) => Promise<void>) {
+  on<TPayload>(event: string, listener: (payload: TPayload) => Promise<void>): void {
```
Comment (B2o5T) [inline packages/services/api/src/modules/shared/providers/message-bus.ts:10]: this method was never `async`, removed `async` keyword

File: packages/services/server/src/graphql-handler.ts [hunk 3]
```diff
@@ -163,15 +163,14 @@ export const graphqlHandler = (options: GraphQLHandlerOptions): RouteHandlerMeth
         });
 
         response.headers.forEach((value, key) => {
-          reply.header(key, value);
+          reply.header(key, value); // eslint-disable-line @typescript-eslint/no-floating-promises -- false positive, FastifyReply.then returns void
```
Comment (B2o5T) [inline packages/services/server/src/graphql-handler.ts:166]: this rule has some issues with `FastifyReply.then` type, more info https://github.com/typescript-eslint/typescript-eslint/issues/4961

File: packages/services/server/src/graphql-handler.ts [hunk 4]
```diff
@@ -163,15 +163,14 @@ export const graphqlHandler = (options: GraphQLHandlerOptions): RouteHandlerMeth
...
         });
 
         if (!reply.hasHeader('x-request-id')) {
-          reply.header('x-request-id', requestId || '');
+          reply.header('x-request-id', requestId || ''); // eslint-disable-line @typescript-eslint/no-floating-promises -- false positive, FastifyReply.then returns void
         }
 
-        reply.status(response.status);
-        reply.send(response.body);
+        reply.status(response.status).send(response.body); // eslint-disable-line @typescript-eslint/no-floating-promises -- false positive, FastifyReply.then returns void
```
Comment (B2o5T) [inline packages/services/server/src/graphql-handler.ts:173]: can be chained

File: packages/services/storage/migrations/migrate.ts [hunk 5]
```diff
@@ -29,17 +29,15 @@ const isDownCommand = cmd === 'down';
...
-      process.exit(1);
-    });
+
+  try {
+    await migrator.up();
+    await migrateClickHouse();
+  } catch (error) {
+    console.error(error);
+    process.exit(1);
+  }
```
Comment (B2o5T) [inline packages/services/storage/migrations/migrate.ts:39]: `Async/Await` is better, and `top-level await` should also be available, also `process.exit(0)` is redundant and can be removed

File: packages/services/usage/src/buffer.ts [hunk 6]
```diff
@@ -254,7 +254,7 @@ export function createKVBuffer<T>(config: {
       const estimatedBufferSize = currentBufferSize + estimatedReportSize;
 
       if (currentBufferSize >= config.limitInBytes || estimatedBufferSize >= config.limitInBytes) {
-        send(true);
+        void send(true);
```
Comment (B2o5T) [inline packages/services/usage/src/buffer.ts:257]: I think in this case we also don't need to `await`, so I added `void`

File: packages/services/usage/src/buffer.ts [hunk 7]
```diff
@@ -269,7 +269,7 @@ export function createKVBuffer<T>(config: {
     } else {
       buffer.push(report);
       if (sumOfOperationsSizeInBuffer() >= config.size) {
-        send(true);
+        void send(true);
```
Comment (B2o5T) [inline packages/services/usage/src/buffer.ts:272]: same

File: packages/services/schema/src/index.ts [hunk 8]
```diff
@@ -21,15 +21,16 @@ async function main() {
...
     name: 'schema',
     tracing: false,
   });
 
   registerShutdown({
     logger: server.log,
     async onShutdown() {
-      await Promise.all([server.close(), redis.disconnect(false)]);
+      await server.close();
+      redis.disconnect(false);
```
Comment (B2o5T) [inline packages/services/schema/src/index.ts:33]: this function do not return `Promise`

File: packages/services/storage/tools/register.cjs [hunk 9]
```diff
@@ -1,3 +0,0 @@
-const { register } = require('esbuild-register/dist/node');
-
-register({ extensions: ['.mjs', '.ts'], format: 'cjs' });
```
Comment (B2o5T) [inline packages/services/storage/tools/register.cjs:3]: I removed this file for the following reasons
1. removing `esbuild-register`
2. removing `strorage/migrations/package.json`
3. removing `storage/tools/register.cjs`
4. exec `db:migrator` script with `node --experimental-specifier-resolution=node --loader ts-node/esm migrations/migrate.ts` as `ts-node` already installed in package.json

**Review by saihaj — APPROVED** [review]

File: packages/libraries/client/src/client.ts [hunk 10]
```diff
@@ -128,8 +128,8 @@ export function createHive(options: HivePluginOptions): HiveClient {
       } else {
         logger.error(`[hive][info] Error ${response.status}: ${response.statusText}`);
       }
-    } catch (error: any) {
-      logger.error(`[hive][info] Error ${error.message}`);
+    } catch (error) {
+      logger.error(`[hive][info] Error ${(error as Error).message}`);
```
[the remark this answers is not in the document]
Comment (B2o5T) [reply packages/libraries/client/src/client.ts:132]: the most of errors are instances of Error

Comment (n1ru4l) [inline packages/libraries/client/src/client.ts:132]:
```suggestion
      logger.error(`[hive][info] Error ${(error as Error)?.message ?? error}`);
```

File: packages/services/storage/tools/register.cjs (hunk 9 above)
Comment (kamilkisiela) [reply packages/services/storage/tools/register.cjs:3]: the thing is, it has to run in docker and as cli, it it works, I'm fine with the changes.

File: packages/libraries/client/src/client.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (B2o5T) [reply packages/libraries/client/src/client.ts:132]: my newly changes don't change previous behaviour of code

**Comment by B2o5T** [comment]
> Can you confirm we can run `migration:create`, `db:generate` and `db` scripts in `@hive/storage`?

@kamilkisiela yes, everything works like before but now code is simpler

**Review by kamilkisiela — APPROVED** [review]

**Pull request merged**